File: Segundo/PDOO/deepSpace/src/deepspace/Damage.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;
import java.util.ArrayList;

/**
 *
 * @author mariacribi
 */
public abstract class Damage {
    int nShields;
    //private  int nWeapons; 
    //private ArrayList <WeaponType> weapons;
    
    //Constructor con visibilidad de paquete
    Damage (int s){
        //nWeapons=w; 
        nShields=s;
        //weapons=null;
    }
    
    //Constructor de copia
    Damage (Damage d){
        nShields=d.nShields;
    }
    
    public abstract DamageToUI getUIversion();
    
    public abstract Damage adjust (ArrayList<Weapon> w, ArrayList<ShieldBooster> s);
     
    public abstract void discardWeapon(Weapon w);
    
    public void discardShieldBooster(){
        if (nShields>0){
            nShields--;
        }
    }
    
    public abstract boolean hasNoEffect();
    
    public int getNShields(){
        return nShields; 
    }
}

File: Segundo/PDOO/ruby p4/Damage.rb
require_relative'DamageToUI'

module Deepspace
    class Damage
        
        def initialize(s)
            @nShields = s
        end

        def getUIversion
            DamageToUI.new(self)
        end

        def to_s
			getUIversion.to_s
		end

        def copy
        end

        def nShields
            @nShields
        end

        def hasNoEffect
        end

        def discardShieldBooster
            if(@nShields>0)
                @nShields=@nShields-1
            else
                @nShields=0
            end
        end

        def discardWeapon(w)
        end

        #array de weapons contiene tal weaponType
        def arrayContainsType(w,t)
            contador=0
            encontrado=false
            while((contador < w.length) && (!encontrado))
                if(w[contador].type==t)
                    encontrado=true
                end
                contador+=1
            end
            
            if(encontrado)
                return contador-1
            else
                return -1
            end
        end

        def adjust(w,s)
            shields=@nShields
            weap=@nWeapons
            if(s.length<@nShields) 
                shields=s.length
            end
        end
        
    end
end
